src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenshotCapture.cs

```diff
@@ -255,13 +255,27 @@ public void WaitForCompletion()
         }
 
         /**
-         * <summary>Calls the onSuccess callback when the readback request finishes or if data is already available</summary>
+         * <summary>
+         * Calls the onSuccess callback when the readback request finishes or if data is already available.
+         * If the system is running in a no graphics mode, this will instead just immediately complete the
+         * action.
+         * </summary>
          */
         public void GetCurrentScreenshotWithCallback(long segmentNumber, Action<(byte[], int, int)?> onCompletion)
         {
             var frame = UnityEngine.Time.frameCount;
             lock (SyncLock)
             {
+
+                // When we don't have graphics available, we immediate invoke the callback with a null value. With how
+                // we use this function now, this needs to happen so the tick data writing task correctly saves the
+                // resulting data.
+                if (SystemInfo.graphicsDeviceType == GraphicsDeviceType.Null && onCompletion != null)
```

Comment:
this if block should be outside the synclock.. we don't need to acquire the lock as we aren't modifying any of the tracking structures